controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,14 +80,582 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+		cephFilesystemSubVolumeGroupDetails.Phase = string(r.cephFilesystemSubVolumeGroup.Status.Phase)
+	}
+
+	r.updateCephResourceStatus(cephFilesystemSubVolumeGroupDetails)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephFilesystemSubVolumeGroup.", "CephFilesystemSubVolumeGroup", klog.KRef(r.cephFilesystemSubVolumeGroup.Namespace, r.cephFilesystemSubVolumeGroup.Name))
+		return err
+	}
+
+	return nil
+
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDProvisioner() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDProvisioner, func() error {
+		if err := r.own(r.cephClientRBDProvisioner); err != nil {
+			return err
+		}
+
+		r.cephClientRBDProvisioner.Spec = desired.Spec
+		return nil
+	})
+
+	cephClientRBDProvisionerDetails := &ocsv1alpha1.CephResourcesSpec{
+		Kind: r.cephClientRBDProvisioner.Kind,
+		Name: r.cephClientRBDProvisioner.Name,
+	}
+	if r.cephClientRBDProvisioner.Status != nil {
+		cephClientRBDProvisionerDetails.Phase = string(r.cephClientRBDProvisioner.Status.Phase)
+	}
+
+	r.updateCephResourceStatus(cephClientRBDProvisionerDetails)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDProvisioner.Namespace, r.cephClientRBDProvisioner.Name))
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRBDNode() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBDNode, func() error {
+		if err := r.own(r.cephClientRBDNode); err != nil {
+			return err
+		}
+
+		r.cephClientRBDNode.Spec = desired.Spec
+		return nil
+	})
+
+	cephClientRBDNodeDetails := &ocsv1alpha1.CephResourcesSpec{
+		Kind: r.cephClientRBDNode.Kind,
+		Name: r.cephClientRBDNode.Name,
+	}
+	if r.cephClientRBDNode.Status != nil {
+		cephClientRBDNodeDetails.Phase = string(r.cephClientRBDNode.Status.Phase)
+	}
+
+	r.updateCephResourceStatus(cephClientRBDNodeDetails)
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBDNode.Namespace, r.cephClientRBDNode.Name))
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientCephFSProvisioner() error {
+
+	// TODO: Update caps relavent to cephfs provisioner
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "allow r",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("allow rw tag cephfs metadata=* path=%s", r.cephFilesystemSubVolumeGroup.Name),
```

Comment:
Rook uses 
```
"mon", "allow r", 
"mgr", "allow rw",
"osd", "allow rw tag cephfs metadata=*"
```

@leseb Can you check the cap changes done is correct for subvolume group integration

Reply:
@ajarr please confirm that the subvolume group integration (`path`) is correctly written. Thanks!

Reply:
The `path=<>` is a part of MDS cap specification, not the OSD. See here:

        https://docs.ceph.com/en/latest/cephfs/client-auth/

e.g.:

        caps: [mds] allow r, allow rw path=/bar

where `path` in your case would be the path to the subvolume or subvolume group as required.